things/service.go

```diff
@@ -147,6 +147,22 @@ type thingsService struct {
 	ulidProvider mainflux.IDProvider
 }
 
+// createUID: take the external id if supplied, otherwise use default provider
+// validate the external id, if validation fails, generate an id using the
+// default provider.
```

Comment:
You can probably simply this to something like
```
provideUUID: Generates UUID or validates external ID if supplied.

Reply:
done